src/delta.cpp

```diff
@@ -405,8 +419,16 @@ Delta generate_diff(const std::string_view &old_data,
         LOG("Emit PATCH relocation @%016x + %016x\n", cur_entry.patched_address,
             new_file_cursor - cur_entry.patched_address);
 #endif
-        patch_data.append(&new_data[cur_entry.patched_address],
-                          new_file_cursor - cur_entry.patched_address);
+        {
```

Comment:
I don't see any smart pointer funny business or RAII at work here, so I'm not super sure why this is in a sub-scope

Reply:
Yeah, I just don't like leaking variables into scope (write_offset, bytes_to_append), especially since this is a direct dupe of the snippet above